miden/src/cli/prove.rs

```diff
@@ -117,7 +134,7 @@ impl ProveCmd {
         );
 
         // write proof to file
-        ProofFile::write(proof, &self.proof_file, &self.assembly_file).map_err(Report::msg)?;
+        ProofFile::write(proof, &self.proof_file, &self.program_file).map_err(Report::msg)?;
```

Comment:
Not from this PR, but for the previous line "Program with hash {} ...", printing the hash twice is redundant. So I would just print `Program proved in {} ms`